### DISCUSSION TITLE
Recently opened workspaces and files

### DISCUSSION THREAD

**Discussion by vzhukovs** [eclipse-theia/theia#7806] [General] [answered]
Hi folks! I'm currently working on these [items](https://github.com/eclipse-theia/theia/issues/4050#issuecomment-558633078), especially on implementing `workbench.action.openRecent`. Want to align [WorkspaceServer](https://github.com/eclipse-theia/theia/blob/80f402c621fe197130eae7abd22a0d89008b4ef1/packages/workspace/src/common/workspace-protocol.ts#L23) interface to be look like in [IWorkspacesService](https://github.com/microsoft/vscode/blob/890f62dfd9f3e70198931f788c5c332b3e8b7ad7/src/vs/platform/workspaces/common/workspaces.ts#L42-L45). But this requires a little modification of related code and tests. 

Methods in [WorkspaceServer](https://github.com/eclipse-theia/theia/blob/80f402c621fe197130eae7abd22a0d89008b4ef1/packages/workspace/src/common/workspace-protocol.ts#L23) are going to be deprecated at the first time, not removed. This is needed to have backward compatibility. 

What do you think of this kind of change?

**Comment by akosyakov** [comment, marked as answer]
It depends on expectation of VS Code extensions. One thing is to provide simple implementation based on existing APIs to support VS Code extensions, another is to reimplement open recent action to support opening recent workspace files, files without workspaces and so on. If you goal is only to support VS Code extension, the latter is not necessary.

Do you know against which extension we can test it? If there is not a real extension it is fine to provide simple implementation based on existing APIs, i.e add something like `WorkspaceService.openRecent` which calls `WorkspaceServer` to fetch recent and then opens it.

**Comment by vzhukovs** [comment]
At this moment I found only one [extension](https://marketplace.visualstudio.com/items?itemName=lfs.vscode-emacs-friendly), that actually calls the command `workbench.action.openRecent` by handling custom hot key. I see. I'll create a simple command handler, that will redirect call to `WorkspaceServer`.
